From 1f93c4d6ef0b7f69ae6a24898c340c19f367da35 Mon Sep 17 00:00:00 2001
From: fang hui <hui.fang@nxp.com>
Date: Wed, 9 Mar 2016 10:34:21 +0800
Subject: [PATCH 45/45] MA-7711 - [testVolumeControl]: The case will cause pop
 exception window

Same on nexus7.
Add code to catch NullPointerException to void pop up exception window.

Signed-off-by: fang hui <hui.fang@nxp.com>
---
 .../src/com/android/systemui/volume/VolumeDialogController.java  | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogController.java b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogController.java
index 32d6805..dec4702 100644
--- a/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogController.java
+++ b/packages/SystemUI/src/com/android/systemui/volume/VolumeDialogController.java
@@ -868,7 +868,14 @@ public class VolumeDialogController {
 
         @Override
         public void onRemoteVolumeChanged(Token token, int flags) {
-            final int stream = mRemoteStreams.get(token);
+            final int stream;
+            try {
+                stream = mRemoteStreams.get(token);
+            } catch (NullPointerException e) {
+                Log.w(TAG, "mRemoteStreams.get(token)", e);
+                return;
+            }
+
             final boolean showUI = (flags & AudioManager.FLAG_SHOW_UI) != 0;
             boolean changed = updateActiveStreamW(stream);
             if (showUI) {
-- 
1.8.0

